cmd/main.go

```diff
@@ -1,136 +1,168 @@
+		Run: runCmd(codes.POST),
 	}
-	help := strings.ToLower(os.Args[1])
-	if help == "-h" || help == "--help" {
-		log.Println(helpMsg)
-		os.Exit(0)
+	postCmd.Flags().StringVarP(&data, "data", "d", "", "Data")
+
+	var deleteCmd = &cobra.Command{
+		Use:   "delete <url>",
+		Short: "Perform a DELETE request on a COAP resource",
+		Example: "coap-cli delete /test -H coap.me -p 5683 -c 50 -d 'hello, world' -O 17,50\n" +
+			"coap-cli delete /test --host coap.me --port 5683 --content-format 50 --data 'hello, world' --options 17,50",
+		Run: runCmd(codes.DELETE),
 	}
+	deleteCmd.Flags().StringVarP(&data, "data", "d", "", "Data")
 
-	code, err := parseCode(strings.ToUpper(os.Args[1]))
-	if err != nil {
-		log.Fatalf("Can't read request code: %s\n%s", err, helpCmd)
+	rootCmd.AddCommand(getCmd, putCmd, postCmd, deleteCmd)
+	rootCmd.PersistentFlags().StringVarP(&host, "host", "H", "localhost", "Host")
+	rootCmd.PersistentFlags().StringVarP(&port, "port", "p", "5683", "Port")
+	rootCmd.PersistentFlags().StringVarP(&auth, "auth", "a", "", "Auth")
+	rootCmd.PersistentFlags().IntVarP(&contentFormat, "content-format", "c", 50, "Content format")
+	rootCmd.PersistentFlags().StringArrayVarP(&options, "options", "O", []string{}, "Options")
+
+	if err := rootCmd.Execute(); err != nil {
+		log.Fatalf("Error executing command: %v", err)
 	}
+}
 
-	if len(os.Args) < 3 {
-		log.Fatalf("CoAP URL must not be empty.\n%s", helpCmd)
+func printMsg(m *pool.Message) {
+	if m != nil {
+		log.Printf("\nMESSAGE:\n%s", m.String())
 	}
-	path := os.Args[2]
-	if strings.HasPrefix(path, "-") {
-		log.Fatalf("Please enter a valid CoAP URL.\n%s", helpCmd)
+	body, err := m.ReadBody()
+	if err != nil {
+		log.Fatalf("failed to read body %v", err)
+	}
+	if len(body) > 0 {
+		log.Printf("\nMESSAGE BODY:\n%s", string(body))
 	}
+}
 
-	os.Args = os.Args[2:]
-	o := flag.Bool("o", false, "Observe")
-	h := flag.String("h", "localhost", "Host")
-	p := flag.String("p", "5683", "Port")
-	// Default type is JSON.
-	cf := flag.Int("cf", 50, "Content format")
-	d := flag.String("d", "", "Message data")
-	a := flag.String("auth", "", "Auth token")
-	flag.Parse()
-
-	client, err := coap.New(*h + ":" + *p)
+func makeRequest(code codes.Code, args []string) {
+	client, err := coap.New(host + ":" + port)
 	if err != nil {
-		log.Fatal("Error creating client: ", err)
+		log.Fatalf("Error coap creating client: %v", err)
 	}
 	var opts coapmsg.Options
-	if a != nil {
-		opts = append(opts, coapmsg.Option{ID: coapmsg.URIQuery, Value: []byte(fmt.Sprintf("auth=%s", *a))})
+	for _, optString := range options {
+		opt := strings.Split(optString, ",")
+		if len(opt) < 2 {
+			log.Fatal("Invalid option format")
+		}
+		optId, err := strconv.ParseUint(opt[0], 10, 16)
+		if err != nil {
+			log.Fatal("Error parsing option id")
+		}
+		opts = append(opts, coapmsg.Option{ID: coapmsg.OptionID(optId), Value: []byte(opt[1])})
 	}
```

Comment:
@1998-felix 
We should have something like hex decoded value, instead of string converted to byte. 


```suggestion
		}
		if strings.HasPrefix(opt[1], "0x") {
			optValue, err := hex.DecodeString(opt[1])
			if err != nil {
				log.Fatal("Invalid option value ", err.Error())
			}
			opts = append(opts, coapmsg.Option{ID: coapmsg.OptionID(optId), Value: optValue})
		} else {
			opts = append(opts, coapmsg.Option{ID: coapmsg.OptionID(optId), Value: []byte(opt[1])})

		}

	}
```